### PR TITLE
tests: reduce the complexity of the test-snapd-sh snap

### PR DESCRIPTION
**Pull request by sergiocazzolato**
To install test-snapd-sh in taking about 40 seconds in a board like pi3.
Whithe this change the istallation takes about 12 seconds.

Also the install local takes as defult path the local test directory, so it is possible to install snaps and is it not needed to pack and install with --dangerous manually.

### CODE REVIEW

File: tests/smoke/sandbox/test-snapd-sh/meta/snap.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: test-snapd-sh
+summary: A no-strings-attached, no-fuss shell for writing tests
+version: 1.0
+
+apps:
+    with-home-plug:
```
Comment (bboozzoo) [inline tests/smoke/sandbox/test-snapd-sh/meta/snap.yaml]: Bunch of the tests assume that there is an app named just like the snap, so `test-snapd-sh` is all you need to run.
```
2019-11-27 23:30:46 Failed tasks: 47
    - google:ubuntu-14.04-64:tests/main/fakestore-install
    - google:ubuntu-14.04-64:tests/main/interfaces-personal-files
    - google:ubuntu-14.04-64:tests/main/interfaces-system-files
    - google:ubuntu-14.04-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-14.04-64:tests/smoke/sandbox
    - google:ubuntu-16.04-32:tests/main/fakestore-install
    - google:ubuntu-16.04-32:tests/main/interfaces-personal-files
    - google:ubuntu-16.04-32:tests/main/interfaces-system-files
    - google:ubuntu-16.04-32:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-16.04-32:tests/main/system-usernames-illegal
    - google:ubuntu-16.04-32:tests/smoke/sandbox
    - google:ubuntu-16.04-64:tests/main/fakestore-install
    - google:ubuntu-16.04-64:tests/main/interfaces-personal-files
    - google:ubuntu-16.04-64:tests/main/interfaces-system-files
    - google:ubuntu-16.04-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-16.04-64:tests/main/system-usernames-illegal
    - google:ubuntu-16.04-64:tests/smoke/sandbox
    - google:ubuntu-18.04-64:tests/main/fakestore-install
    - google:ubuntu-18.04-64:tests/main/interfaces-personal-files
    - google:ubuntu-18.04-64:tests/main/interfaces-system-files
    - google:ubuntu-18.04-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-18.04-64:tests/main/system-usernames-illegal
    - google:ubuntu-18.04-64:tests/smoke/sandbox
    - google:ubuntu-19.04-64:tests/main/fakestore-install
    - google:ubuntu-19.04-64:tests/main/interfaces-personal-files
    - google:ubuntu-19.04-64:tests/main/interfaces-system-files
    - google:ubuntu-19.04-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-19.04-64:tests/main/system-usernames-illegal
    - google:ubuntu-19.04-64:tests/smoke/sandbox
    - google:ubuntu-19.10-64:tests/main/fakestore-install
    - google:ubuntu-19.10-64:tests/main/interfaces-personal-files
    - google:ubuntu-19.10-64:tests/main/interfaces-system-files
    - google:ubuntu-19.10-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-19.10-64:tests/main/system-usernames-illegal
    - google:ubuntu-19.10-64:tests/smoke/sandbox
    - google:ubuntu-core-16-64:tests/main/fakestore-install
    - google:ubuntu-core-16-64:tests/main/interfaces-personal-files
    - google:ubuntu-core-16-64:tests/main/interfaces-system-files
    - google:ubuntu-core-16-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-core-16-64:tests/main/system-usernames-illegal
    - google:ubuntu-core-16-64:tests/smoke/sandbox
    - google:ubuntu-core-18-64:tests/main/fakestore-install
    - google:ubuntu-core-18-64:tests/main/interfaces-personal-files
    - google:ubuntu-core-18-64:tests/main/interfaces-system-files
    - google:ubuntu-core-18-64:tests/main/parallel-install-common-dirs-undo
    - google:ubuntu-core-18-64:tests/main/system-usernames-illegal
    - google:ubuntu-core-18-64:tests/smoke/sandbox
2019-11-27 23:30:46 Failed task prepare: 8
    - google:ubuntu-14.04-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-16.04-32:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-16.04-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-18.04-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-19.04-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-19.10-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-core-16-64:tests/main/interfaces-snapd-control-with-manage
    - google:ubuntu-core-18-64:tests/main/interfaces-snapd-control-with-manage
error: unsuccessful run
```

in this case the snap meta should probably be:
```
name: test-snapd-sh
summary: A no-strings-attached, no-fuss shell for writing tests
version: 1.0

apps:
    test-snapd-sh:
        command: bin/sh
        plugs: [home]
````

Comment (sergiocazzolato) [reply tests/smoke/sandbox/test-snapd-sh/meta/snap.yaml]: @bboozzoo All the issues were fixed, thanks for the review

File: tests/main/interfaces-personal-files/task.yaml [hunk 2]
```diff
@@ -46,9 +46,6 @@ execute: |
     echo "When the interface is connected"
     snap connect test-snapd-sh:personal-files
 
-    echo "And not other interfaces are connected"
```
Comment (stolowski) [inline tests/main/interfaces-personal-files/task.yaml]: Why this change? I think it's useful to test that without home this snap can still access personal files, or am I missing something?

File: tests/main/interfaces-system-files/task.yaml [hunk 3]
```diff
@@ -46,9 +46,6 @@ execute: |
     echo "When the interface is connected"
     snap connect test-snapd-sh:system-files
 
-    echo "And not other interfaces are connected"
```
Comment (stolowski) [inline tests/main/interfaces-system-files/task.yaml]: Same question.

File: tests/main/interfaces-broadcom-asic-control/test-snapd-sh/meta/snap.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: test-snapd-sh
+summary: A no-strings-attached, no-fuss shell for writing tests
```
Comment (zyga) [inline tests/main/interfaces-broadcom-asic-control/test-snapd-sh/meta/snap.yaml]: The summary here (and everywhere) is no more accurate. I guess we can drop it entirely?

**Comment by zyga** [comment]
In general, this is nice, I really like the direction and removal of generalization and reuse of extremely heavy snap like test-snapd-sh (note that it used to say "no fuss shell"). I will review the change carefully next week.

File: tests/main/interfaces-broadcom-asic-control/test-snapd-sh/meta/snap.yaml (hunk 4 above)
Comment (stolowski) [reply tests/main/interfaces-broadcom-asic-control/test-snapd-sh/meta/snap.yaml]: We discussed this on standup and the agreement is to update all the names (including snap names) and summaries in a followup to reduce the diff (changing names would affect task.yamls).

File: tests/main/interfaces-personal-files/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-personal-files/task.yaml]: This line was included previously because the test-snapd-sh snap was plugging the home interface and by default it is automatic, so we needed to disconnect manually, but with this change the test-snapd-sh snap is not plugging the home interface anymore so this is not needed, in face the test was failing with this line :)

File: tests/main/interfaces-system-files/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-system-files/task.yaml]: same answer :)

**Comment by sergiocazzolato** [comment]
@stolowski @zyga thanks for the review!!!

File: tests/main/interfaces-personal-files/task.yaml (hunk 2 above)
Comment (stolowski) [reply tests/main/interfaces-personal-files/task.yaml]: Right, makes sense, thanks.

File: tests/lib/snaps.sh [hunk 5]
```diff
@@ -2,26 +2,32 @@
...
 make_snap() {
     local SNAP_NAME="$1"
-    shift;
-    local SNAP_FILE="$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
-    local SNAP_DIR
+    local SNAP_DIR="$TESTSLIB/snaps/${SNAP_NAME}"
+    if [ $# -gt 1 ]; then
+        SNAP_DIR="$2"
+    fi
+    local SNAP_FILE="${SNAP_DIR}/${SNAP_NAME}_1.0_all.snap"
```
Comment (zyga) [inline tests/lib/snaps.sh]: I didn't before realise the "1.0" version is relevant.

**Comment by anonymouse64** [comment]
The store was responding with 503s so the failed with maximum output uploaded to travis while running tests/main/refresh-delta. 
I have seen this test fail before numerous times with maximum output, perhaps there is something we can do to prevent this test from not producing megabytes of `taskrunner.go:426: DEBUG: Running task 55 on Doing: Consider re-refresh of "test-snapd-delta-refresh"`?

Anyways, I have restarted that suite on Travis
